src/PrestaShopBundle/Form/Admin/Improve/Shipping/Carrier/Type/CarrierRangesCostsType.php

```diff
@@ -0,0 +1,79 @@
+<?php
+/**
+ * Copyright since 2007 PrestaShop SA and Contributors
+ * PrestaShop is an International Registered Trademark & Property of PrestaShop SA
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.md.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+
+declare(strict_types=1);
+
+namespace PrestaShopBundle\Form\Admin\Improve\Shipping\Carrier\Type;
+
+use PrestaShopBundle\Form\Admin\Improve\Shipping\Carrier\Type\Costs\CostsZoneType;
+use PrestaShopBundle\Form\Admin\Type\TranslatorAwareType;
+use Symfony\Component\Form\Extension\Core\Type\CollectionType;
+use Symfony\Component\Form\FormBuilderInterface;
+use Symfony\Component\OptionsResolver\OptionsResolver;
+
+/**
+ * CarrierRangesCostsType is a form type used to set prices by zones and ranges
+ *
+ * $builder
+ *     ->add('ranges', CarrierRangesCostsType::class)
+ * ;
+ */
+class CarrierRangesCostsType extends TranslatorAwareType
```

Comment:
If the form type only contains one collection then it's probably overkill, you could add this collection directly in `ShippingLocationsAndCostsType` This could remove an extra layer in the form data structure making it simpler